types/compare.go

```diff
@@ -111,6 +111,18 @@ func CompareFloat64(x, y float64) int {
 	return 1
 }
 
+// CompareFloat64Fixed returns an integer comparing the float64 x to y.
+// It treat two float64 is equal if the difference of them less than eps.
+func CompareFloat64Fixed(x, y, eps float64) int {
+	if x < y {
+		return -1
+	} else if x == y || math.Abs(x-y) < eps {
+		return 0
+	}
+
```

Comment:
Check equal first,  what if `x < y && math.Abs(x-y) < eps` ?

Reply:
Regardless of `x < y` or `x > y`, if `abs(x - y) < eps` we think they are equal under the error specified by `eps`.

Reply:
I mean that `x < y && math.Abs(x-y) < eps` will retrun -1 in this code.
should be:
if x == y || math.Abs(x-y) < eps {
return 0
}
if x < y {
return -1
}
return 1